image_classification/tensorflow/official/resnet/resnet_run_loop.py

```diff
@@ -166,13 +166,64 @@ def learning_rate_fn(global_step):
         warmup_steps, tf.float32))
     return tf.cond(global_step < warmup_steps, lambda: warmup_lr, lambda: lr)
 
+  def poly_rate_fn(global_step):
+    """Handles linear scaling rule, gradual warmup, and LR decay.
+
+    The learning rate starts at 0, then it increases linearly per step.  After
+    flags.poly_warmup_epochs, we reach the base learning rate (scaled to account
+    for batch size). The learning rate is then decayed using a polynomial rate
+    decay schedule with power 2.0.
+
+    Args:
+    global_step: the current global_step
+
+    Returns:
+    returns the current learning rate
+    """
+
+    # Learning rate schedule for LARS polynomial schedule
+    if flags.batch_size < 8192:
```

Comment:
You'll need to pass in batch size. (flags is not a global)

Reply:
`batch_size` is passed as a function parameter here.